modules/xfem/src/base/XFEMGeometricCut3D.C

```diff
@@ -0,0 +1,168 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "XFEMGeometricCut3D.h"
+
+#include "MooseError.h"
+#include "XFEMFuncs.h"
+
+XFEMGeometricCut3D::XFEMGeometricCut3D(Real t0, Real t1):
+  XFEMGeometricCut(t0,t1),
+  _center(Point(0.0,0.0,0.0)),
+  _normal(Point(0.0,0.0,0.0))
+{}
+
+XFEMGeometricCut3D::~XFEMGeometricCut3D()
+{}
+
+bool
+XFEMGeometricCut3D::cutElementByGeometry(const Elem* /*elem*/,
+                                         std::vector<cutEdge> & /*cutEdges*/,
+                                         Real /*time*/)
+{
+  mooseError("invalid method for 3D mesh cutting");
+  return false;
+}
+
+bool
+XFEMGeometricCut3D::cutElementByGeometry(const Elem* elem,
+                                         std::vector<cutFace> & cutFaces,
+                                         Real /*time*/)
+//TODO: Time evolving cuts not yet supported in 3D (hence the lack of use of the time variable)
+{
+  bool cut_elem = false;
+
+   int hex_indices[6][4] = {{0,3,2,1},{0,1,5,4},{1,2,6,5},{2,3,7,6},{3,0,4,7},{4,5,6,7}};
+   int tet_indices[4][3] = {{0,2,1},{0,1,3},{1,2,3},{2,0,3}};
+
+   unsigned int n_face_nodes = 0;
+   if (elem->n_nodes() == 8)
+     n_face_nodes = 4;
+   else if (elem->n_nodes() == 4)
+     n_face_nodes = 3;
+   else
+     mooseError("this method only works for lienar hexes and tets");
```

Comment:
I agree completely.  There's a method on the fe class that will tell us that, but not on the elem.  Any idea on how to get the fe associated with an elem when that's all we have in our hand?